### PR TITLE
 add a test for subqueries

### PR DESCRIPTION
**Pull request by jellegerbrandy**

### CODE REVIEW

File: test/cache.spec.ts [hunk 1]
```diff
@@ -128,13 +128,44 @@ describe('apolloClient caching checks', () => {
...
+          voter
+          proposal {
+            id
+          }
+          outcome
+          reputation
+        }
+      `
+    const xx = await arc.sendQuery(qry)
+    console.log(xx.data.proposal.votes.length)
```
Comment (tibetsprague) [inline test/cache.spec.ts]: remove

File: test/cache.spec.ts [hunk 2]
```diff
@@ -92,7 +92,7 @@ describe('apolloClient caching checks', () => {
     await new Member(member.id as string , arc).state({ fetchPolicy: 'cache-only'}).pipe(first()).toPromise()
   })
 
-  it('pre-fetching ProposalVotes works', async () => {
+  it.only('pre-fetching ProposalVotes works', async () => {
```
Comment (tibetsprague) [inline test/cache.spec.ts]: why .only?

Comment (jellegerbrandy) [reply test/cache.spec.ts]: still work in progress :-)

File: src/proposal.ts [hunk 3]
```diff
@@ -210,9 +210,7 @@ export class Proposal implements IStateful<IProposalState> {
       totalRepWhenExecuted
       title
       url
-      votes {
-        id
-      }
+      # votes { id }
```
Comment (tibetsprague) [inline src/proposal.ts]: is this commenting this out?

File: src/proposal.ts [hunk 4]
```diff
@@ -457,7 +471,7 @@ export class Proposal implements IStateful<IProposalState> {
       } else {
         throw Error(`Unknown proposal type or incomplete proposal data`)
       }
-      // the  formule to enter into the preboosted state is:
+      // the  formule to enter into the preboosted ttate is:
```
Comment (tibetsprague) [inline src/proposal.ts]: misspell

File: test/cache.spec.ts [hunk 5]
```diff
@@ -110,74 +91,55 @@ describe('apolloClient caching checks', () => {
...
+    arc.getObservable(query, { subscribe: true, fetchPolicy: 'no-cache'}).subscribe((x: any) => {
       subscribed = true
+      results.push(x)
     })
     await waitUntilTrue(() => subscribed)
-
-    // we now get all proposal data without hitting the cache
-    const proposalData = await proposal.state({ fetchPolicy: 'cache-only'}).pipe(first()).toPromise()
-    expect(proposalData.scheme.id).toEqual(scheme.id)
+    // const proposalData = await proposal.state().pipe(first()).toPromise()
```
Comment (tibetsprague) [inline test/cache.spec.ts]: remove

**Pull request merged**